src/extractive.py

```diff
@@ -953,32 +955,43 @@ def test_epoch_end(self, outputs):
         for name, value in rouge_scores_log.items():
             self.log(name, value, prog_bar=False)
 
-    def predict(self, input_text, raw_scores=False, num_summary_sentences=3):
-        """Summaries ``input_text`` using the model.
+    def predict_sentences(
+        self, input_sentences: Union[List[str], types.GeneratorType], raw_scores=False, num_summary_sentences=3, tokenized=False
+    ):
+        """Summarizes ``input_sentences`` using the model.
 
         Args:
-            input_text (str): The text to be summarized.
-            raw_scores (bool, optional): Return a dictionary containing each sentence
+            input_sentences (list or generator): The sentences to be summarized.
+            raw_scores (bool, optional): Return a list containing each sentence
                 and its corespoding score instead of the summary. Defaults to False.
             num_summary_sentences (int, optional): The number of sentences in the
                 output summary. This value specifies the number of top sentences to
                 select as the summary. Defaults to 3.
+            tokenized (bool, optional): If the input sentences are already tokenized
+                using spacy. If true, ``input_sentences`` should be a list of lists
+                where the outer list contains sentences and the inner lists contain
+                tokens. Defaults to False.
 
         Returns:
-            str: The summary text. If ``raw_scores`` is set then returns a dictionary
+            str: The summary text. If ``raw_scores`` is set then returns a list
             of input sentences and their corespoding scores.
         """
-        nlp = English()
-        sentencizer = nlp.create_pipe("sentencizer")
-        nlp.add_pipe(sentencizer)
-        doc = nlp(input_text)
-
         # Create source text.
         # Don't add periods when joining because that creates a space before the period.
-        src_txt = [
-            " ".join([token.text for token in sentence if str(token) != "."]) + "."
-            for sentence in doc.sents
-        ]
+        if tokenized:
+            src_txt = [
+                " ".join([token.text for token in sentence if str(token) != "."]) + "."
```

Comment:
Shouldn't it be `token` here instead of `token.text` or are we assuming it will be a spacy Token object.

Reply:
I was thinking we assume its a spacy token object. I've added to the docstring to clarify this.